docs/connector-development/connector-specification-reference.md

```diff
@@ -43,17 +41,132 @@ For example, using the following spec:
 }
 ```
 
-will result in the following configuration on the UI: 
+will result in the following configuration on the UI:
 
 ![Screen Shot 2021-11-18 at 7 14 04 PM](https://user-images.githubusercontent.com/6246757/142558797-135f6c73-f05d-479f-9d88-e20cae85870c.png)
 
-
 :::info
 
-Within an object definition, if some fields have the `order` property defined, and others don't, then the fields without the `order` property defined should be rendered last in the UI. Among those elements (which don't have `order` defined), no ordering is guaranteed. 
+Within an object definition, if some fields have the `order` property defined, and others don't, then the fields without the `order` property defined should be rendered last in the UI. Among those elements (which don't have `order` defined), no ordering is guaranteed.
 
 :::
 
+### Collapsing optional fields
+
+By default, all optional fields will be collapsed into an `Optional fields` section which can be expanded or collapsed by the user. This helps streamline the UI for setting up a connector by initially focusing attention on the required fields only.
```

Comment:
should we add a few words here that optional fields are moved to the bottom of the form/group if they don't have an explicit order specified?